### PR TITLE
Update to make chromebrew work on Google OnHub.

### PR DESCRIPTION
**Pull request by snailium**
1. Added installing "xz" package, because "xz" is not available on OnHub.

2. Updated install script to use curl instead of wget, becuase wget is not available on OnHub.

### CODE REVIEW

**Comment by jam7** [comment]
Thank you for the PR.  We are discussing about the use of curl in #882.  Please wait for a while until we make a decision.

File: install.sh [hunk 1]
```diff
@@ -156,15 +160,15 @@ done
...
 chmod +x crew
 rm -f $CREW_PREFIX/bin/crew
 ln -s `pwd`/crew $CREW_PREFIX/bin
 #install crew library
 mkdir -p $CREW_LIB_PATH/lib
 cd $CREW_LIB_PATH/lib
-wget -N $URL/lib/package.rb
-wget -N $URL/lib/package_helpers.rb
+curl -R $URL/lib/package.rb
+curl -R $URL/lib/package_helpers.rb
```
Comment (jam7) [inline install.sh]: What you need here is `curl -o package.rb -z package.rb $URL/lib/package.rb`.

File: install.sh [hunk 2]
```diff
@@ -78,7 +82,7 @@ function download_check () {
 
     #download
     echo "Downloading $1..."
-    wget -c $2 -O $3
+    curl --ignore-content-length -# -L --ssl $2 -o "$3"
```
Comment (jam7) [inline install.sh]: what you need is `curl -C` not `--ignore-content-length`

File: crew [hunk 3]
```diff
@@ -370,7 +370,7 @@ def download
     sha256sum = @pkg.binary_sha256[@device[:architecture]]
   end
   Dir.chdir CREW_BREW_DIR do
-    system('wget', '--continue', '--no-check-certificate', url, '-O', filename)
+    system('curl', '--insecure', '-L', '-#', url, '-o', filename)
```
Comment (jam7) [inline crew]: you need `-C` here also.

File: install.sh (hunk 2 above)
Comment (jam7) [inline install.sh]: sry, it is `curl -C -`

File: crew (hunk 3 above)
Comment (snailium) [inline crew]: I tried ```-C -``` for this one, but I got error for the second dash.

**Comment by jam7** [comment]
did you try like ..., '-C', '-', ...?  if so, onhub may have different
vesion of curl?

2017年8月2日(水) 10:09 snailium <notifications@github.com>:

**Comment by snailium** [comment]
Here is the OnHub's curl version.

```
chronos@localhost /home/chronos $ crew remove curl
Package curl isn't installed.
chronos@localhost /home/chronos $ which curl
/usr/bin/curl
chronos@localhost /home/chronos $ curl --version
curl 7.51.0 (armv7a-cros-linux-gnueabi) libcurl/7.51.0 OpenSSL/1.0.2j zlib/1.2.8 c-ares/1.12.0
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile NTLM SSL libz TLS-SRP UnixSockets 
```

File: crew (hunk 3 above)
Comment (snailium) [inline crew]: OK, here is the detailed error for ```-C -```

```
curl: option -C -: expected a proper numerical parameter
curl: try 'curl --help' or 'curl --manual' for more information
```

**Comment by flyingP0tat0** [comment]
As for general interest, always use the `-L` option to let `curl` follow links like `wget`. Won't work otherwise with this, as GitHub provides links you need to follow.

**Comment by jam7** [comment]
It works like this on Chrombe OS.  The version of curl is identical to OnHub.
```
chronos@localhost ~/armv7l $ crew remove curl
Package curl isn't installed.
chronos@localhost ~/armv7l $ type -a curl
curl is /usr/bin/curl
chronos@localhost ~/armv7l $ curl --version
curl 7.51.0 (armv7a-cros-linux-gnueabi) libcurl/7.51.0 OpenSSL/1.0.2l zlib/1.2.11
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smtp smtps telnet tftp 
Features: Largefile NTLM SSL libz TLS-SRP UnixSockets 
chronos@localhost ~/armv7l $ curl -L -C - -# https://github.com/jam7/chrome-cross/releases/download/v1.8/linux-headers-3.18-chromeos-armv7l.tar.xz -o test.tar.xz
######################################################################## 100.0%
chronos@localhost ~/armv7l $ curl -L -C - -# https://github.com/jam7/chrome-cross/releases/download/v1.8/linux-headers-3.18-chromeos-armv7l.tar.xz -o test.tar.xz
######################################################################## 100.0%
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
```

**Comment by jam7** [comment]
@flyingP0tat0 that's true.  we need `-L` for every curl.

**Comment by snailium** [comment]
It seems running in bash is fine.

```
chronos@localhost /home/chronos $ curl -L -C - -# https://github.com/jam7/chrome-cross/releases/download/v1.8/linux-headers-3.18-chromeos-armv7l.tar.xz -o test.tar.xz
######################################################################## 100.0%
```

But not in Ruby.
```ruby
    system('curl', '-C -', '--insecure', '-L', '-#', url, '-o', filename)
```
I got
```
Precompiled binary available, downloading...
curl: option -C -: expected a proper numerical parameter
curl: try 'curl --help' or 'curl --manual' for more information
```

Tried
```ruby
   system('curl', '-C', '-', '--insecure', '-L', '-#', url, '-o', filename)
```
is working OK. So the trick is separating the ```'-c -'``` to ```'-c', '-'```

**Comment by snailium** [comment]
Hi @jam7 , I've corrected the curl commands.

File: crew [hunk 4]
```diff
@@ -10,7 +10,7 @@ require 'fileutils'
 @pkgName = ARGV[1]
 
 ARCH = `uname -m`.strip
-ARCH_LIB = if ARCH == 'x86_64' then 'lib64' else 'lib' end
+ARCH_LIB = if ARCH == 'x86_64' or ARCH == 'aarch64' then 'lib64' else 'lib' end
```
Comment (jam7) [inline crew]: Is this modification required for OnHub?

For the case of chromebook, aarch64 is 64 bit kernel but 32 bit userland.  So, we need to use `lib` for aarch64.

**Comment by jam7** [comment]
@snailium Do you have a plan to fix above last problem?  If no, I'll modify it if it is OK to you.  Please let us know.  Thanks.

**Pull request merged**